test/appium/tests/atomic/chats/test_chats_management.py

```diff
@@ -0,0 +1,124 @@
+        sign_in_view = SignInView(self.driver)
+        sign_in_view.create_user()
+        home_view = sign_in_view.get_home_view()
+        home_view.add_contact(recipient['public_key'])
+        chat_view = home_view.get_chat_view()
+        for _ in range(4):
+            chat_view.chat_message_input.send_keys('test message')
+            chat_view.send_message_button.click()
+        chat_view.clear_history()
+        if not chat_view.no_messages_in_chat.is_element_present():
+            pytest.fail('Message history is shown')
+        home_view.relogin()
+        home_view.get_chat_with_user(recipient['username']).click()
+        if not chat_view.no_messages_in_chat.is_element_present():
+            pytest.fail('Message history is shown after re-login')
+
+    @marks.testrail_id(1395)
+    def test_swipe_to_delete_1_1_chat(self):
+        sign_in = SignInView(self.driver)
+        home = sign_in.create_user()
+        chat = home.add_contact(group_chat_users['A_USER']['public_key'])
+        chat.chat_message_input.send_keys('test message')
+        chat.send_message_button.click()
+        chat.get_back_to_home_view()
+        home.get_chat_with_user(group_chat_users['A_USER']['username']).swipe_and_delete()
+        self.driver.close_app()
+        self.driver.launch_app()
+        sign_in.accept_agreements()
+        sign_in.sign_in()
+        if home.get_chat_with_user(group_chat_users['A_USER']['username']).is_element_displayed():
+            pytest.fail('Deleted 1-1 chat is present after relaunch app')
+
+    @marks.testrail_id(3718)
+    def test_swipe_to_delete_public_chat(self):
+        sign_in = SignInView(self.driver)
+        home = sign_in.create_user()
+        chat_name = home.get_public_chat_name()
+        chat = home.join_public_chat(chat_name)
+        message = 'test message'
+        chat.chat_message_input.send_keys(message)
+        chat.send_message_button.click()
+        chat.get_back_to_home_view()
+        home.get_chat_with_user('#' + chat_name).swipe_and_delete()
+        profile = home.profile_button.click()
+        profile.logout()
+        sign_in.click_account_by_position(0)
+        sign_in.sign_in()
+        if home.get_chat_with_user('#' + chat_name).is_element_displayed():
+            self.errors.append('Deleted public chat is present after relogin')
+        home.join_public_chat(chat_name)
+        if chat.chat_element_by_text(message).is_element_displayed():
+            self.errors.append('Chat history is shown')
+        self.verify_no_errors()
+
+    @marks.testrail_id(3694)
+    def test_add_contact_from_public_chat(self):
+        sign_in = SignInView(self.driver)
+        home = sign_in.create_user()
+        chat_name = 'testaddcontact'
+        chat = home.join_public_chat(chat_name)
+        message = 'test message'
+
+        chat.connection_status.wait_for_invisibility_of_element(30)
+        chat_element = chat.chat_element_by_text(message)
```

Comment:
please, redefine `click` method in `ChatElementByText` in order to click on `Tap to reconnect..` button (if present) for avoiding failures like https://saucelabs.com/beta/tests/d12495d1bce94e36a07a60f3f18aff6a/commands?auth=bfc00b8f2dbecbe96774aa3cf8721eb8#48